rpc/transport/transport.go

```diff
@@ -55,37 +58,32 @@ type Transport interface {
 	Close() error
 }
 
-type OutgoingMessage struct {
-	Message  rpccp.Message
-	send     func() error
-	released bool
-}
-
-func (o *OutgoingMessage) Release() {
-	if m := o.Message.Message(); !o.released && m != nil {
-		m.Release()
-	}
-}
-
-func (o *OutgoingMessage) Send() error {
-	if !o.released {
-		return o.send()
-	}
-
-	panic("Tried to send() a message that was already released.")
-}
-
-type IncomingMessage struct {
-	Message     rpccp.Message
-	ReleaseHook capnp.ReleaseFunc
+// OutgoingMessage is a message that can be sent at a later time.
+// Release() MUST be called when the OutgoingMessage is no longer in
+// use. Before releasing an ougoing message, Send() MAY be called at
+// most once to send the message over the transport that produced it.
+//
+// Implementations MUST ensure the cap table is empty before Send is
```

Comment:
I think this MUST is incorrect; in fact, we keep the cap table full in the case of sending Return messages, in case we need to pipeline later questions on the result.